cmd/kubeadm/app/kubeadm.go

```diff
@@ -69,6 +69,17 @@ func Run() error {
 	pflag.CommandLine.MarkHidden("google-json-key")
 	pflag.CommandLine.MarkHidden("log-flush-frequency")
 
+	if err := checkPermissions(); !err {
+		return err
+	}
+
 	cmd := cmd.NewKubeadmCommand(cmdutil.NewFactory(nil), os.Stdin, os.Stdout, os.Stderr, getEnvParams())
 	return cmd.Execute()
 }
+
+func checkPermissions() error {
+	if uid := os.Getuid(); uid != 0 {
+		return fmt.Errorf("Kubeadm needs to run as uid `0`. It is being run as %d", uid)
```

Comment:
There are a few issues with this error message, first of all it's not exactly user-friendly to talk about UIDs. We should suggest either "run as root" or "use sudo kubeadm".

Secondly, please never capitalise the `kubeadm`, and try to adhere to the rest of error message, we have `<$package/$basename> $message` convention right now.

I'd make it something like:
```
fmt.Errorf("<kubeadm> unable to run as non-root user")
```
The rest of errors come from packages under `app`, so prefix here defers from the convention, but let's discuss. I could be `<main>`, I suppose... We can discuss and improve the convention, of course, but for now let's not break the UX we have.